### PR TITLE
Added options parameter to rest api functions.

### PR DESCRIPTION
**Pull request by antdjohns**
### Summary
Adding options param to all endpoints to allow for additional request options. This change is being added to support EdgeHeaders that will be needed for Launch pad.

RV1:
- Added `options: Optional[dict]` to all endpoints leading to BaseClient.get.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -12,18 +12,19 @@
...
+os.environ["POLYGON_API_KEY"] = "POLYGON_API_KEY"
 
 # -- Project information -----------------------------------------------------
 
-project = 'polygon-api-client'
-copyright = '2022, Polygon.io'
-author = 'Polygon.io'
+project = "polygon-api-client"
+copyright = "2022, Polygon.io"
+author = "Polygon.io"
```
Comment (jbonzo) [inline docs/source/conf.py:24]: Can we not make any changes to this file. It doesn't seem related to the proposed changes.

File: polygon/rest/aggs.py [hunk 2]
```diff
@@ -78,15 +81,19 @@ def get_grouped_daily_aggs(
             result_key="results",
             deserializer=GroupedDailyAgg.from_dict,
             raw=raw,
+            options=options,
         )
 
+    # TODO: next breaking change release move "market_type" to be 2nd mandatory
+    # param
```
Comment (jbonzo) [inline polygon/rest/aggs.py:88]: I don't think this comment is related to the function below.

**Review by jbonzo — COMMENTED** [review]
I think you also want to update the docstring

File: polygon/rest/aggs.py (hunk 2 above)
Comment (antdjohns) [reply polygon/rest/aggs.py:88]: this was added due to a line move.

File: docs/source/conf.py (hunk 1 above)
Comment (antdjohns) [reply docs/source/conf.py:24]: this is from running poetry run black, its auto formatting.

File: polygon/rest/aggs.py (hunk 2 above)
Comment (antdjohns) [reply polygon/rest/aggs.py:88]: Im not sure why it's showing as an addition without a deduction. I didn't add this line. I can rollback if needed.

**Comment by antdjohns** [comment]
Closing this request as Edge Headers pr covers these changes
